src/store/auth.js

```diff
@@ -1,59 +1,67 @@
-import vueAuthInstance from '../services/auth.js'
+import {
+  initVueAuthenticate
+} from '../services/auth.js'
+
+let vueAuthInstance
 
 export default {
   state: {
-    user: {
-      displayname: '',
-      email: 'x'
-    },
-    isAuthenticated: vueAuthInstance.isAuthenticated()
+    displayname: '',
+    email: 'x',
+    isAuthenticated: false
+  },
+  getters: {
+    isAuthenticated: state => {
+      return state.isAuthenticated
+    }
   },
-
   mutations: {
-    isAuthenticated (state, payload) {
+    IS_AUTHENTICATED(state, payload) {
       state.isAuthenticated = payload.isAuthenticated
     },
-    setUser (state, payload) {
+    SET_USER(state, payload) {
       state.user = payload
     }
   },
 
   actions: {
-    login (context, payload) {
+    login(context, payload) {
       payload = payload || {}
-      return vueAuthInstance.login(payload.user, payload.requestOptions).then(function () {
-        context.commit('isAuthenticated', {
+      return vueAuthInstance.login(payload.user, payload.requestOptions).then(function() {
+        context.commit('IS_AUTHENTICATED', {
           isAuthenticated: vueAuthInstance.isAuthenticated()
         })
       })
     },
 
-    logout (context, payload) {
+    logout(context, payload) {
       payload = payload || {}
-      return vueAuthInstance.logout(payload.requestOptions).then(function () {
-        context.commit('isAuthenticated', {
+      return vueAuthInstance.logout(payload.requestOptions).then(function() {
+        context.commit('IS_AUTHENTICATED', {
           isAuthenticated: vueAuthInstance.isAuthenticated()
         })
-        context.commit('setUser', {
+        context.commit('SET_USER', {
           displayname: '',
           email: ''
         })
       })
     },
 
-    initAuth (context, payload) {
+    initAuth(context, payload) {
       this._vm.$client.loginWithBearer(vueAuthInstance.getToken()).then(res => {
-        context.commit('setUser', {
+        context.commit('SET_USER', {
           displayname: res['display-name'],
           email: Object.keys(res.email).length === 0 ? '' : res.email
         })
       })
     },
 
-    authenticate (context, payload) {
+    authenticate(context, payload) {
+      // TODO shouldn't we use vue-authenticate global & inject it into vue ?
+      vueAuthInstance = initVueAuthenticate(context.rootState.config.auth)
```

Comment:
context.rootState.config.auth is undefined ....

![screenshot from 2018-11-05 14-43-07](https://user-images.githubusercontent.com/1005065/48001596-531a7c00-e109-11e8-8a34-c49892b50f47.png)